### PR TITLE
Debug page; create table and join table

### PR DESCRIPTION
**Pull request by jonah-katz**
Screen Shot 2020-12-20 at 11 05 49 PM

### CODE REVIEW

**Comment by jacklieblich** [comment]
Ready for review?

**Comment by jonah-katz** [comment]
Sure!

**Comment by jacklieblich** [comment]
Wanna update the naming in `table.ts` as per whatsapp convo?

File: nodejs/src/classes/deck.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default class Deck {
+  cards: Array<number>;
```
Comment (jacklieblich) [inline nodejs/src/classes/deck.ts:2]: How's this gonna work?

Comment (jacklieblich) [inline nodejs/src/classes/deck.ts:2]: How's this gonna work?

**Comment by jonah-katz** [comment]
> Wanna update the naming in `table.ts` as per whatsapp convo?

absolutely

File: nodejs/src/classes/table.ts [hunk 2]
```diff
@@ -1,20 +1,77 @@
...
+};
+
+enum PlayerAction {
+  SmallBlind,
+  BigBlind,
+  Call,
+  Raise,
+  Check,
+  Fold,
+  AllIn
```
Comment (jacklieblich) [inline nodejs/src/classes/table.ts:23]: Why's this action different than a raise/call?

File: nodejs/src/classes/table.ts [hunk 3]
```diff
@@ -1,20 +1,77 @@
...
-  constructor(json: any) {
-    this.small_blind = json.small_blind;
-    this.big_blind = json.big_blind;
+  constructor() {
+    this.table_id = 'j8dj';
+    this.small_blind = 1;
+    this.big_blind = 2;
+    this.players = [];
+    this.game = null;
   }
```
Comment (jacklieblich) [inline nodejs/src/classes/table.ts:52]: How's this work?

File: nodejs/src/classes/deck.ts (hunk 1 above)
Comment (jonah-katz) [reply nodejs/src/classes/deck.ts:2]: Haven't fully figured it out, but thinking shuffling and dealing occurs in this class. Can do things like:

 Deck.reset() -> fresh shuffled 52
 Deck.pop2() -> take 2 cards out
Deck.pop3() -> take 3 cards out

etc

File: nodejs/src/classes/table.ts (hunk 2 above)
Comment (jonah-katz) [reply nodejs/src/classes/table.ts:23]: Still thinking this through, but thinking theres a distinct button on the app for "all in". Also, we'd want to indicate that a user went all in.

File: nodejs/src/classes/table.ts (hunk 3 above)
Comment (jonah-katz) [reply nodejs/src/classes/table.ts:52]: This is how you set the default variable values upon instantiating a new instance of the class

Comment (jonah-katz) [reply nodejs/src/classes/table.ts:52]: The game (renamed to hand) will get set later, when its time to actually deal. You can imagine the table gets made once somebody click create table in the app.. folks gradually join in... but the hand actually gets made once somebody hits "deal"

File: nodejs/src/classes/deck.ts (hunk 1 above)
Comment (jonah-katz) [reply nodejs/src/classes/deck.ts:2]: Each Table has 1 Deck

Comment (jacklieblich) [reply nodejs/src/classes/deck.ts:2]: how is a number gonna represent a card? order cards 1-52 or something?

File: nodejs/src/classes/table.ts (hunk 3 above)
Comment (jacklieblich) [reply nodejs/src/classes/table.ts:52]: Cool. Why's table_id and blinds hardcoded?

Comment (jonah-katz) [reply nodejs/src/classes/table.ts:52]: Just for now

File: nodejs/src/classes/deck.ts (hunk 1 above)
Comment (jonah-katz) [reply nodejs/src/classes/deck.ts:2]: Yes exactly. 1-13 = spades, 14-26 = clubs...

File: nodejs/src/classes/table.ts (hunk 3 above)
Comment (jonah-katz) [reply nodejs/src/classes/table.ts:52]: Blinds will be hardcoded until we implement custom table settings

**Review by jacklieblich — APPROVED** [review]
🤙

**Pull request merged**